{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:23:11) \\n[GCC 9.4.0]'"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#CONDA ENV base_conda (python 3.9.7)\n",
    "#Import packages\n",
    "#---------------------------------------\n",
    "import sys\n",
    "import os\n",
    "import glob\n",
    "import pandas as pd\n",
    "from matplotlib import pyplot as plt\n",
    "import numpy as np\n",
    "from multiprocessing import Pool\n",
    "from tqdm import tqdm\n",
    "\n",
    "\n",
    "#Import your modules\n",
    "#---------------------------------------\n",
    "import te_rna_f as ter\n",
    "sys.path.insert(1, '/cndd3/dburrows/CODE/admin_tools/')\n",
    "from admin_tools import admin_functions as adm\n",
    "\n",
    "# Define paths\n",
    "#----------------------------------------------------------------------\n",
    "l_code = '/Users/dominicburrows/Dropbox/PhD/Analysis/my_scripts/GitHub/'\n",
    "l_data = '/Users/dominicburrows/Dropbox/PhD/analysis/Project/'\n",
    "l_fig = '/Users/dominicburrows/Dropbox/PhD/figures/'\n",
    "\n",
    "s_code = '/cndd3/dburrows/CODE/'\n",
    "s_data = '/cndd3/dburrows/DATA/'\n",
    "s_fig = '/cndd3/dburrows/FIGS/'\n",
    "\n",
    "%load_ext autoreload\n",
    "sys.version"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make new directories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "curr = glob.glob('/cndd/dburrows/DATA/te/rna/PE.counts/ATEM_old/Sample*')\n",
    "curr = [os.path.basename(i) for i in curr]\n",
    "os.chdir('/cndd/dburrows/DATA/te/rna/PE.counts/ATEM/')\n",
    "for c in curr:\n",
    "    os.mkdir(c)\n",
    "\n",
    "# in_path= \"/cndd3/dburrows/DATA/te/rna/PE.bam/\"\n",
    "# out_path = \"/cndd/dburrows/DATA/te/rna/PE.counts/TET/\"\n",
    "\n",
    "# samp_l = glob.glob(in_path+'/*Samp*')\n",
    "\n",
    "# for s in samp_l:\n",
    "#     #find base directory of s\n",
    "#     base = os.path.basename(s)\n",
    "#     if not os.path.exists(out_path+'/'+base):\n",
    "#         os.mkdir(out_path+'/'+base)\n",
    "# #5, 8, 9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "#symlink files \n",
    "curr = glob.glob(\"/cndd/dburrows/DATA/te/rna/PE.bam_multi20/\" + '*Samp*')\n",
    "nex = glob.glob('/datasets/Public_Datasets/Dracheva_PsychEncode_development/processed/rna_seq/' + '*-*')\n",
    "\n",
    "for c in curr:\n",
    "    short = os.path.basename(c).split('_')[-1]\n",
    "    prac = [i for i in nex if short in i]\n",
    "    if len(prac) == 1:\n",
    "         os.symlink(prac[0] + '/Aligned.sortedByCoord.out.bam', c + '/Aligned.sortedByCoord.out.bam')\n",
    "    elif len(prac) < 1:\n",
    "        short = short.upper()\n",
    "        prac = [i for i in nex if short in i]\n",
    "        if len(prac) == 1:\n",
    "            os.symlink(prac[0] + '/Aligned.sortedByCoord.out.bam', c + '/Aligned.sortedByCoord.out.bam')\n",
    "        else:\n",
    "            if short == 'HCT16HECA028-GABA':\n",
    "                short = 'HCK16HECA028-GABA'\n",
    "            if short == 'HCT16HECA028-GLU':\n",
    "                short = 'HCK16HECA028-GLU'\n",
    "            prac = [i for i in nex if short in i]\n",
    "            os.symlink(prac[0] + '/Aligned.sortedByCoord.out.bam', c + '/Aligned.sortedByCoord.out.bam')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Delete all files in directory\n",
    "os.chdir('/cndd3/dburrows/DATA/te/rna/CZI.counts/ATEM/pseudobulk')\n",
    "curr = glob.glob('**')\n",
    "for c in curr:\n",
    "    #remove all contents of each directory\n",
    "    shutil.rmtree(c)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check for log files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "os.chdir('/cndd/dburrows/DATA/te/rna/PE.counts/ATEM/')\n",
    "curr_ = glob.glob('*Sa*')\n",
    "for c in curr_:\n",
    "    os.chdir(c)\n",
    "    atem = glob.glob('/cndd/dburrows/DATA/te/rna/PE.counts/ATEM/' + c + '/*ATEM*')\n",
    "    log = glob.glob('/cndd/dburrows/DATA/te/rna/PE.counts/ATEM/' + c + '/*log*')\n",
    "    if not atem or not log:\n",
    "        print(c)\n",
    "    if log:\n",
    "        if 'All reads accounted for' not in np.load(log[0]):\n",
    "            print(c)\n",
    "\n",
    "    os.chdir('..')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "'All reads accounted for' in np.load(log[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Save config files for snakemake"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "my_dict = {\n",
    "    \"in_path\": \"/cndd3/dburrows/DATA/te/rna/PE.bam\", \n",
    "    \"out_path\": \"/cndd/dburrows/DATA/te/rna/PE.counts/ATEM\",\n",
    "    \"bed_plus_path\": \"/cndd3/dburrows/DATA/annotations/rmsk/rmsk.hg38.filt-5ptrim.plus.full.sort.bed\",\n",
    "    \"bed_minus_path\": \"/cndd3/dburrows/DATA/annotations/rmsk/rmsk.hg38.filt-5ptrim.minus.full.sort.bed\"\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Write successful\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import glob\n",
    "os.chdir('/cndd/dburrows/DATA/te/rna/PE.counts/ATEM')\n",
    "curr_dict = my_dict.copy()\n",
    "curr_dict['ID'] = glob.glob('*Samp*GABA*')\n",
    "adm.make_config(curr_dict, '/cndd3/dburrows/CODE/te_rna_ageing/snakemake/workflow_PE-GABA.ATEM/config.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'in_path': '/cndd3/dburrows/DATA/te/rna/PE.bam',\n",
       " 'out_path': '/cndd/dburrows/DATA/te/rna/PE.counts/ATEM',\n",
       " 'bed_plus_path': '/cndd3/dburrows/DATA/annotations/rmsk/rmsk.hg38.filt-5ptrim.plus.full.sort.bed',\n",
       " 'bed_minus_path': '/cndd3/dburrows/DATA/annotations/rmsk/rmsk.hg38.filt-5ptrim.minus.full.sort.bed',\n",
       " 'ID': ['Sample_5077_GABA',\n",
       "  'Sample_1275_GLU',\n",
       "  'Sample_5086_GABA',\n",
       "  'Sample_4379_GLU',\n",
       "  'Sample_5570_GLU',\n",
       "  'Sample_5918_GABA',\n",
       "  'Sample_5744_GLU',\n",
       "  'Sample_4337_GABA',\n",
       "  'Sample_5184_GABA',\n",
       "  'Sample_HCT15HBMA007_GABA',\n",
       "  'Sample_4337_GLU',\n",
       "  'Sample_198035-1765_GABA',\n",
       "  'Sample_5977_GLU',\n",
       "  'Sample_4428_GABA',\n",
       "  'Sample_5643_GABA',\n",
       "  'Sample_6285_GLU',\n",
       "  'Sample_5976_GABA',\n",
       "  'Sample_3739_GLU',\n",
       "  'Sample_4425_GLU',\n",
       "  'Sample_1277_GABA',\n",
       "  'Sample_1134_GLU',\n",
       "  'Sample_5570_GABA',\n",
       "  'Sample_1133_GLU',\n",
       "  'Sample_6285_GABA',\n",
       "  'Sample_1823_GABA',\n",
       "  'Sample_5161_GABA',\n",
       "  'Sample_198035-1765_GLU',\n",
       "  'Sample_179134-1401_GABA',\n",
       "  'Sample_4321_GLU',\n",
       "  'Sample_5643_GLU',\n",
       "  'Sample_5326_GABA',\n",
       "  'Sample_4411_GABA',\n",
       "  'Sample_1539_GLU',\n",
       "  'Sample_4414_GABA',\n",
       "  'Sample_HCTZZNA014_GLU',\n",
       "  'Sample_1133_GABA',\n",
       "  'Sample_4369_GABA',\n",
       "  'Sample_5977_GABA',\n",
       "  'Sample_4725_GLU',\n",
       "  'Sample_198034-1435_GABA',\n",
       "  'Sample_5976_GLU',\n",
       "  'Sample_5387_GLU',\n",
       "  'Sample_4425_GABA',\n",
       "  'Sample_6007_GABA',\n",
       "  'Sample_HCT17HEIA010_GABA',\n",
       "  'Sample_1241_GLU',\n",
       "  'Sample_5446_GABA',\n",
       "  'Sample_5871_GABA',\n",
       "  'Sample_5936_GLU',\n",
       "  'Sample_4545_GABA',\n",
       "  'Sample_HCT17HEIA010_GLU',\n",
       "  'Sample_HCT15HBMA007_GLU',\n",
       "  'Sample_6032_GABA',\n",
       "  'Sample_HCT16HECA028_GABA',\n",
       "  'Sample_179134-1401_GLU',\n",
       "  'Sample_4332_GABA',\n",
       "  'Sample_4379_GABA',\n",
       "  'Sample_5871_GLU',\n",
       "  'Sample_4411_GLU',\n",
       "  'Sample_HCT15HAWA031_GABA',\n",
       "  'Sample_1848_GABA',\n",
       "  'Sample_5309_GLU',\n",
       "  'Sample_198034-1435_GLU',\n",
       "  'Sample_HCT16HCQA020_GLU',\n",
       "  'Sample_4369_GLU',\n",
       "  'Sample_5744_GABA',\n",
       "  'Sample_5401_GABA',\n",
       "  'Sample_5086_GLU',\n",
       "  'Sample_5293_GABA',\n",
       "  'Sample_HCT16HECA028_GLU',\n",
       "  'Sample_1848_GLU',\n",
       "  'Sample_5387_GABA',\n",
       "  'Sample_1241_GABA',\n",
       "  'Sample_1648_GABA',\n",
       "  'Sample_5606_GLU',\n",
       "  'Sample_HCT15HAWA031_GLU',\n",
       "  'Sample_5918_GLU',\n",
       "  'Sample_5617_GABA',\n",
       "  'Sample_HCTYPA008_GABA',\n",
       "  'Sample_5617_GLU',\n",
       "  'Sample_5161_GLU',\n",
       "  'Sample_4332_GLU',\n",
       "  'Sample_5936_GABA',\n",
       "  'Sample_5451_GLU',\n",
       "  'Sample_1277_GLU',\n",
       "  'Sample_HCTZZNA014_GABA',\n",
       "  'Sample_4725_GABA',\n",
       "  'Sample_HCT16HCQA020_GABA',\n",
       "  'Sample_5077_GLU',\n",
       "  'Sample_HCTYPA008_GLU',\n",
       "  'Sample_4428_GLU',\n",
       "  'Sample_5446_GLU',\n",
       "  'Sample_1648_GLU',\n",
       "  'Sample_5787_GLU',\n",
       "  'Sample_4545_GLU',\n",
       "  'Sample_5401_GLU',\n",
       "  'Sample_5451_GABA',\n",
       "  'Sample_4414_GLU',\n",
       "  'Sample_5787_GABA',\n",
       "  'Sample_1105_GLU',\n",
       "  'Sample_4321_GABA',\n",
       "  'Sample_1275_GABA',\n",
       "  'Sample_5606_GABA',\n",
       "  'Sample_1823_GLU',\n",
       "  'Sample_HCT15HBNA032_GLU',\n",
       "  'Sample_1539_GABA',\n",
       "  'Sample_5293_GLU',\n",
       "  'Sample_5184_GLU',\n",
       "  'Sample_5309_GABA',\n",
       "  'Sample_6032_GLU',\n",
       "  'Sample_5326_GLU',\n",
       "  'Sample_HCT15HBNA032_GABA',\n",
       "  'Sample_1105_GABA',\n",
       "  'Sample_1535_GLU',\n",
       "  'Sample_6007_GLU',\n",
       "  'Sample_1134_GABA']}"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\"bed_minus_path\": \"/cndd3/dburrows/DATA/annotations/rmsk/rmsk.hg38.filt-5ptrim.minus.full.sort.bed\", \"ID\": [\"Sample_5086-GLU\", \"Sample_4369-GLU\", \"Sample_5309-GLU\", \"Sample_4411-GLU\", \"Sample_5871-GLU\", \"Sample_5936-GLU\", \"Sample_1241-GLU\", \"Sample_5387-GLU\", \"Sample_5976-GLU\", \"Sample_4725-GLU\", \"Sample_1539-GLU\", \"Sample_Hct16HCQA020-GLU\", \"Sample_5643-GLU\", \"Sample_4321-GLU\", \"Sample_1133-GLU\", \"Sample_1134-GLU\", \"Sample_4425-GLU\", \"Sample_3739-GLU\", \"Sample_6285-GLU\", \"Sample_5977-GLU\", \"Sample_4337-GLU\", \"Sample_HctZZNA014-GLU\", \"Sample_5744-GLU\", \"Sample_5570-GLU\", \"Sample_4379-GLU\", \"Sample_Hct15HBMA007-GLU\", \"Sample_1275-GLU\", \"Sample_Hct17HEIA010-GLU\", \"Sample_HctYPA008-GLU\", \"Sample_6007-GLU\", \"Sample_1535-GLU\", \"Sample_5326-GLU\", \"Sample_6032-GLU\", \"Sample_198034-1435-GLU\", \"Sample_5184-GLU\", \"Sample_5293-GLU\", \"Sample_1823-GLU\", \"Sample_Hct16HECA028-GLU\", \"Sample_1105-GLU\", \"Sample_4414-GLU\", \"Sample_179134-1401-GLU\", \"Sample_5401-GLU\", \"Sample_4545-GLU\", \"Sample_5787-GLU\", \"Sample_1648-GLU\", \"Sample_5446-GLU\", \"Sample_4428-GLU\", \"Sample_Hct15HAWA031-GLU\", \"Sample_Hct15HBNA032-GLU\", \"Sample_5077-GLU\", \"Sample_1277-GLU\", \"Sample_5451-GLU\", \"Sample_4332-GLU\", \"Sample_198035-1765-GLU\", \"Sample_5161-GLU\", \"Sample_5617-GLU\", \"Sample_5918-GLU\", \"Sample_5606-GLU\", \"Sample_1848-GLU\"]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'in_path': '/cndd/dburrows/DATA/te/rna/PE.bam_multi20',\n",
       " 'out_path': '/cndd/dburrows/DATA/te/rna/PE.counts/RNA-mut',\n",
       " 'ID': ['Sample_5570-GABA',\n",
       "  'Sample_5086-GLU',\n",
       "  'Sample_1823-GABA',\n",
       "  'Sample_6285-GABA',\n",
       "  'Sample_1277-GABA',\n",
       "  'Sample_5976-GABA',\n",
       "  'Sample_5161-GABA',\n",
       "  'Sample_4411-GABA',\n",
       "  'Sample_4369-GLU',\n",
       "  'Sample_5326-GABA',\n",
       "  'Sample_HctYPA008-GABA',\n",
       "  'Sample_5309-GLU',\n",
       "  'Sample_4411-GLU',\n",
       "  'Sample_5871-GLU',\n",
       "  'Sample_5077-GABA',\n",
       "  'Sample_198034-1435-GABA',\n",
       "  'Sample_5086-GABA',\n",
       "  'Sample_Hct15HBNA032-GABA',\n",
       "  'Sample_5936-GLU',\n",
       "  'Sample_4337-GABA',\n",
       "  'Sample_1241-GLU',\n",
       "  'Sample_5184-GABA',\n",
       "  'Sample_5918-GABA',\n",
       "  'Sample_5643-GABA',\n",
       "  'Sample_5387-GLU',\n",
       "  'Sample_5976-GLU',\n",
       "  'Sample_4725-GLU',\n",
       "  'Sample_4428-GABA',\n",
       "  'Sample_1539-GLU',\n",
       "  'Sample_4332-GABA',\n",
       "  'Sample_4379-GABA',\n",
       "  'Sample_1848-GABA',\n",
       "  'Sample_179134-1401-GABA',\n",
       "  'Sample_Hct16HCQA020-GLU',\n",
       "  'Sample_5643-GLU',\n",
       "  'Sample_4321-GLU',\n",
       "  'Sample_5744-GABA',\n",
       "  'Sample_5401-GABA',\n",
       "  'Sample_5293-GABA',\n",
       "  'Sample_1133-GLU',\n",
       "  'Sample_1134-GLU',\n",
       "  'Sample_4425-GLU',\n",
       "  'Sample_3739-GLU',\n",
       "  'Sample_6285-GLU',\n",
       "  'Sample_6007-GABA',\n",
       "  'Sample_4425-GABA',\n",
       "  'Sample_1133-GABA',\n",
       "  'Sample_4414-GABA',\n",
       "  'Sample_5977-GLU',\n",
       "  'Sample_5977-GABA',\n",
       "  'Sample_Hct16HCQA020-GABA',\n",
       "  'Sample_4369-GABA',\n",
       "  'Sample_4337-GLU',\n",
       "  'Sample_HctZZNA014-GLU',\n",
       "  'Sample_5744-GLU',\n",
       "  'Sample_5446-GABA',\n",
       "  'Sample_5871-GABA',\n",
       "  'Sample_5570-GLU',\n",
       "  'Sample_6032-GABA',\n",
       "  'Sample_4379-GLU',\n",
       "  'Sample_Hct15HBMA007-GLU',\n",
       "  'Sample_1275-GLU',\n",
       "  'Sample_Hct17HEIA010-GLU',\n",
       "  'Sample_4545-GABA',\n",
       "  'Sample_198035-1765-GABA',\n",
       "  'Sample_HctYPA008-GLU',\n",
       "  'Sample_5617-GABA',\n",
       "  'Sample_6007-GLU',\n",
       "  'Sample_1535-GLU',\n",
       "  'Sample_Hct16HECA028-GABA',\n",
       "  'Sample_5936-GABA',\n",
       "  'Sample_5326-GLU',\n",
       "  'Sample_Hct17HEIA010-GABA',\n",
       "  'Sample_6032-GLU',\n",
       "  'Sample_198034-1435-GLU',\n",
       "  'Sample_5184-GLU',\n",
       "  'Sample_5293-GLU',\n",
       "  'Sample_1823-GLU',\n",
       "  'Sample_4725-GABA',\n",
       "  'Sample_Hct16HECA028-GLU',\n",
       "  'Sample_1105-GLU',\n",
       "  'Sample_4414-GLU',\n",
       "  'Sample_179134-1401-GLU',\n",
       "  'Sample_5401-GLU',\n",
       "  'Sample_4545-GLU',\n",
       "  'Sample_1648-GABA',\n",
       "  'Sample_5787-GLU',\n",
       "  'Sample_1648-GLU',\n",
       "  'Sample_5446-GLU',\n",
       "  'Sample_4428-GLU',\n",
       "  'Sample_1241-GABA',\n",
       "  'Sample_5387-GABA',\n",
       "  'Sample_Hct15HAWA031-GABA',\n",
       "  'Sample_Hct15HAWA031-GLU',\n",
       "  'Sample_Hct15HBNA032-GLU',\n",
       "  'Sample_5077-GLU',\n",
       "  'Sample_1539-GABA',\n",
       "  'Sample_1275-GABA',\n",
       "  'Sample_4321-GABA',\n",
       "  'Sample_5606-GABA',\n",
       "  'Sample_1277-GLU',\n",
       "  'Sample_Hct15HBMA007-GABA',\n",
       "  'Sample_5309-GABA',\n",
       "  'Sample_1134-GABA',\n",
       "  'Sample_5451-GLU',\n",
       "  'Sample_1105-GABA',\n",
       "  'Sample_4332-GLU',\n",
       "  'Sample_198035-1765-GLU',\n",
       "  'Sample_5161-GLU',\n",
       "  'Sample_5617-GLU',\n",
       "  'Sample_HctZZNA014-GABA',\n",
       "  'Sample_5918-GLU',\n",
       "  'Sample_5606-GLU',\n",
       "  'Sample_1848-GLU',\n",
       "  'Sample_5787-GABA',\n",
       "  'Sample_5451-GABA']}"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('5451-GABA_ATGGCATG-GGTACCTT_HKVTWDSX3', '5451-GABA')]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[(curr[i], p) for i in range(len(curr)) if p in curr[i]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5570-GABA\n",
      "[('5570-GABA_TCTCTACT-GAACCGCG_HKVTWDSX3', '5570-GABA')]\n",
      "5086-GLU\n",
      "[('5086-GLU_TCTGTTGG-CCATTCGA_HKVTWDSX3', '5086-GLU')]\n",
      "1823-GABA\n",
      "[('1823-GABA_GATCTATC-AGCCTCAT_HKVTWDSX3', '1823-GABA')]\n",
      "6285-GABA\n",
      "[('6285-GABA_ACTAAGAT-CCGCGGTT_HKVTWDSX3', '6285-GABA')]\n",
      "1277-GABA\n",
      "[('1277-GABA_AAGATACT-ATGTAAGT_HKVTWDSX3', '1277-GABA')]\n",
      "5976-GABA\n",
      "[('5976-GABA_GCCACAGG-ATGGCATG_HKVTWDSX3', '5976-GABA')]\n",
      "5161-GABA\n",
      "[('5161-GABA_AGGTTATA-CGGAACTG_HKVTWDSX3', '5161-GABA')]\n",
      "4411-GABA\n",
      "[('4411-GABA_CAATTAAC-ATATCTCG_HKVTWDSX3', '4411-GABA')]\n",
      "4369-GLU\n",
      "[('4369-GLU_TAAGGTCA-CTACGACA_HKVTWDSX3', '4369-GLU')]\n",
      "5326-GABA\n",
      "[('5326-GABA_ATGTAAGT-CATAGAGT_HKVTWDSX3', '5326-GABA')]\n",
      "HctYPA008-GABA\n",
      "[]\n",
      "5309-GLU\n",
      "[('5309-GLU_GCGCAAGC-CGGCGTGA_HKVTWDSX3', '5309-GLU')]\n",
      "4411-GLU\n",
      "[('4411-GLU_TGGCCGGT-GCGCTCTA_HKVTWDSX3', '4411-GLU')]\n",
      "5871-GLU\n",
      "[('5871-GLU_ACGCACCT-CCTTCACC_HKVTWDSX3', '5871-GLU')]\n",
      "5077-GABA\n",
      "[('5077-GABA_ACACTAAG-ATATGGAT_HKVTWDSX3', '5077-GABA')]\n",
      "198034-1435-GABA\n",
      "[('198034-1435-GABA_TATCGCAC-ACACTAAG_HKVTWDSX3', '198034-1435-GABA')]\n",
      "5086-GABA\n",
      "[('5086-GABA_CTCACCAA-TTGCCTAG_HKVTWDSX3', '5086-GABA')]\n",
      "Hct15HBNA032-GABA\n",
      "[('Hct15HBNA032-GABA_TAAGTGGT-GGCTTAAG_HKVTWDSX3', 'Hct15HBNA032-GABA')]\n",
      "5936-GLU\n",
      "[('5936-GLU_GTCTACAC-ACCTTGGC_HKVTWDSX3', '5936-GLU')]\n",
      "4337-GABA\n",
      "[('4337-GABA_CCGCGGTT-CTAGCGCT_HKVTWDSX3', '4337-GABA')]\n",
      "1241-GLU\n",
      "[('174051-1241-GLU_GGTCACGA-CATAATAC_HKVTWDSX3', '1241-GLU')]\n",
      "5184-GABA\n",
      "[('5184-GABA_TATCGCAC-ACACTAAG_HKVTWDSX3', '5184-GABA')]\n",
      "5918-GABA\n",
      "[('5918-GABA_AGTACTCC-AACAGGTT_HKVTWDSX3', '5918-GABA')]\n",
      "5643-GABA\n",
      "[('5643-GABA_AGGCAGAG-GGCATTCT_HKVTWDSX3', '5643-GABA')]\n",
      "5387-GLU\n",
      "[('5387-GLU_TGGTGGCA-TTACAGGA_HKVTWDSX3', '5387-GLU')]\n",
      "5976-GLU\n",
      "[('5976-GLU_ATTGTGAA-GCAATGCA_HKVTWDSX3', '5976-GLU')]\n",
      "4725-GLU\n",
      "[('4725-GLU_CATAGAGT-TGGTGGCA_HKVTWDSX3', '4725-GLU')]\n",
      "4428-GABA\n",
      "[('4428-GABA_ACAGGCGC-AGGCAGAG_HKVTWDSX3', '4428-GABA')]\n",
      "1539-GLU\n",
      "[('1539-GLU_GCTTGTCA-GTATGTTC_HKVTWDSX3', '1539-GLU')]\n",
      "4332-GABA\n",
      "[('4332-GABA_ACAGGCGC-AGGCAGAG_HKVTWDSX3', '4332-GABA')]\n",
      "4379-GABA\n",
      "[('4379-GABA_CTCACCAA-TTGCCTAG_HKVTWDSX3', '4379-GABA')]\n",
      "1848-GABA\n",
      "[('1848-GABA_ATATCTCG-ACTAAGAT_HKVTWDSX3', '1848-GABA')]\n",
      "179134-1401-GABA\n",
      "[('179134-1401-GABA_TGCGGCGT-TACCGAGG_HKVTWDSX3', '179134-1401-GABA')]\n",
      "Hct16HCQA020-GLU\n",
      "[]\n",
      "5643-GLU\n",
      "[('5643-GLU_GAATGAGA-AATGCCTC_HKVTWDSX3', '5643-GLU')]\n",
      "4321-GLU\n",
      "[('4321-GLU_CCATTCGA-CGGACAAC_HKVTWDSX3', '4321-GLU')]\n",
      "5744-GABA\n",
      "[('5744-GABA_CTGCTTCC-GATCTATC_HKVTWDSX3', '5744-GABA')]\n",
      "5401-GABA\n",
      "[('5401-GABA_TAATACAG-GTGAATAT_HKVTWDSX3', '5401-GABA')]\n",
      "5293-GABA\n",
      "[('5293-GABA_AGGTTATA-CGGAACTG_HKVTWDSX3', '5293-GABA')]\n",
      "1133-GLU\n",
      "[('1133-GLU_CGTTAGAA-GACCTGAA_HKVTWDSX3', '1133-GLU')]\n",
      "1134-GLU\n",
      "[('1134-GLU_GCAGAATT-TGGCCGGT_HKVTWDSX3', '1134-GLU')]\n",
      "4425-GLU\n",
      "[('4425-GLU_CTACGACA-TTGGACTC_HKVTWDSX3', '4425-GLU')]\n",
      "3739-GLU\n",
      "[('3739-GLU_CTAGCGCT-GTCTACAC_HKVTWDSX3', '3739-GLU')]\n",
      "6285-GLU\n",
      "[('6285-GLU_GTCGGAGC-TTATAACC_HKVTWDSX3', '6285-GLU')]\n",
      "6007-GABA\n",
      "[('6007-GABA_TACTCATA-GCCACAGG_HKVTWDSX3', '6007-GABA')]\n",
      "4425-GABA\n",
      "[('4425-GABA_TCGTAGTG-CCAAGTCT_HKVTWDSX3', '4425-GABA')]\n",
      "1133-GABA\n",
      "[('1133-GABA_TACCGAGG-AGTTCAGG_HKVTWDSX3', '1133-GABA')]\n",
      "4414-GABA\n",
      "[('4414-GABA_AGTTCAGG-TCTGTTGG_HKVTWDSX3', '4414-GABA')]\n",
      "5977-GLU\n",
      "[('5977-GLU_AAGTCCAA-TACTCATA_HKVTWDSX3', '5977-GLU')]\n",
      "5977-GABA\n",
      "[('5977-GABA_GGACTTGG-CGTCTGCG_HKVTWDSX3', '5977-GABA')]\n",
      "Hct16HCQA020-GABA\n",
      "[]\n",
      "4369-GABA\n",
      "[('4369-GABA_CGGAACTG-TCGTAGTG_HKVTWDSX3', '4369-GABA')]\n",
      "4337-GLU\n",
      "[('4337-GLU_TTATAACC-TCGATATC_HKVTWDSX3', '4337-GLU')]\n",
      "HctZZNA014-GLU\n",
      "[]\n",
      "5744-GLU\n",
      "[('5744-GLU_TCATCCTT-AGCTCGCT_HKVTWDSX3', '5744-GLU')]\n",
      "5446-GABA\n",
      "[('5446-GABA_GTTCCAAT-GCAGAATT_HKVTWDSX3', '5446-GABA')]\n",
      "5871-GABA\n",
      "[('5871-GABA_GTATGTTC-TTCCTGTT_HKVTWDSX3', '5871-GABA')]\n",
      "5570-GLU\n",
      "[('5570-GLU_ATGAGGCC-CAATTAAC_HKVTWDSX3', '5570-GLU')]\n",
      "6032-GABA\n",
      "[('6032-GABA_CTGCTTCC-GATCTATC_HKVTWDSX3', '6032-GABA')]\n",
      "4379-GLU\n",
      "[('4379-GLU_TCTGTTGG-CCATTCGA_HKVTWDSX3', '4379-GLU')]\n",
      "Hct15HBMA007-GLU\n",
      "[]\n",
      "1275-GLU\n",
      "[('1275-GLU_TGGATCGA-TATCGCAC_HKVTWDSX3', '1275-GLU')]\n",
      "Hct17HEIA010-GLU\n",
      "[('Hct17HEIA010-GLU_GATTCTGC-CTCTCGTC_HKVTWDSX3', 'Hct17HEIA010-GLU')]\n",
      "4545-GABA\n",
      "[('4545-GABA_CTTGGTAT-GGACTTGG_HKVTWDSX3', '4545-GABA')]\n",
      "198035-1765-GABA\n",
      "[('198035-1765-GABA_TCGATATC-ACTCGTGT_HKVTWDSX3', '198035-1765-GABA')]\n",
      "HctYPA008-GLU\n",
      "[]\n",
      "5617-GABA\n",
      "[('5617-GABA_CCGTGAAG-ATCCACTG_HKVTWDSX3', '5617-GABA')]\n",
      "6007-GLU\n",
      "[('6007-GLU_CGTCTGCG-ATTGTGAA_HKVTWDSX3', '6007-GLU')]\n",
      "1535-GLU\n",
      "[('1535-GLU_TCGATATC-ACTCGTGT_HKVTWDSX3', '1535-GLU')]\n",
      "Hct16HECA028-GABA\n",
      "[]\n",
      "5936-GABA\n",
      "[('5936-GABA_ACTCGTGT-GTTCCAAT_HKVTWDSX3', '5936-GABA')]\n",
      "5326-GLU\n",
      "[('5326-GLU_GCACGGAC-TGCGAGAC_HKVTWDSX3', '5326-GLU')]\n",
      "Hct17HEIA010-GABA\n",
      "[('Hct17HEIA010-GABA_AGCCTCAT-TCTCTACT_HKVTWDSX3', 'Hct17HEIA010-GABA')]\n",
      "6032-GLU\n",
      "[('6032-GLU_TCATCCTT-AGCTCGCT_HKVTWDSX3', '6032-GLU')]\n",
      "198034-1435-GLU\n",
      "[('198034-1435-GLU_CGCTATGT-GTGTCGGA_HKVTWDSX3', '198034-1435-GLU')]\n",
      "5184-GLU\n",
      "[('5184-GLU_CGCTATGT-GTGTCGGA_HKVTWDSX3', '5184-GLU')]\n",
      "5293-GLU\n",
      "[('5293-GLU_GAACCGCG-TAAGGTCA_HKVTWDSX3', '5293-GLU')]\n",
      "1823-GLU\n",
      "[('1823-GLU_AGCTCGCT-GATTCTGC_HKVTWDSX3', '1823-GLU')]\n",
      "4725-GABA\n",
      "[('4725-GABA_TGCGAGAC-CAACAATG_HKVTWDSX3', '4725-GABA')]\n",
      "Hct16HECA028-GLU\n",
      "[]\n",
      "1105-GLU\n",
      "[('1105-GLU_GGTCACGA-CATAATAC_HKVTWDSX3', '1105-GLU')]\n",
      "4414-GLU\n",
      "[('4414-GLU_GACCTGAA-CTCACCAA_HKVTWDSX3', '4414-GLU')]\n",
      "179134-1401-GLU\n",
      "[('179134-1401-GLU_CATAATAC-CGTTAGAA_HKVTWDSX3', '179134-1401-GLU')]\n",
      "5401-GLU\n",
      "[('5401-GLU_CGGCGTGA-ACAGGCGC_HKVTWDSX3', '5401-GLU')]\n",
      "4545-GLU\n",
      "[('4545-GLU_TCCAACGC-AAGTCCAA_HKVTWDSX3', '4545-GLU')]\n",
      "1648-GABA\n",
      "[('1648-GABA_AACAGGTT-CTTGGTAT_HKVTWDSX3', '1648-GABA')]\n",
      "5787-GLU\n",
      "[('5787-GLU_CCTTCACC-GGAGCGTC_HKVTWDSX3', '5787-GLU')]\n",
      "1648-GLU\n",
      "[('1648-GLU_GGTGAACC-TCCAACGC_HKVTWDSX3', '1648-GLU')]\n",
      "5446-GLU\n",
      "[('5446-GLU_ACCTTGGC-ATGAGGCC_HKVTWDSX3', '5446-GLU')]\n",
      "4428-GLU\n",
      "[('4428-GLU_GTGAATAT-GAATGAGA_HKVTWDSX3', '4428-GLU')]\n",
      "1241-GABA\n",
      "[('174051-1241-GABA_AACTGTAG-TGCGGCGT_HKVTWDSX3', '1241-GABA')]\n",
      "5387-GABA\n",
      "[('5387-GABA_CAACAATG-CCGTGAAG_HKVTWDSX3', '5387-GABA')]\n",
      "Hct15HAWA031-GABA\n",
      "[]\n",
      "Hct15HAWA031-GLU\n",
      "[]\n",
      "Hct15HBNA032-GLU\n",
      "[('Hct15HBNA032-GLU_CGGACAAC-AATCCGGA_HKVTWDSX3', 'Hct15HBNA032-GLU')]\n",
      "5077-GLU\n",
      "[('5077-GLU_GTGTCGGA-GCGCAAGC_HKVTWDSX3', '5077-GLU')]\n",
      "1539-GABA\n",
      "[('1539-GABA_ATCCACTG-ACGCACCT_HKVTWDSX3', '1539-GABA')]\n",
      "1275-GABA\n",
      "[('1275-GABA_AACGTTCC-AGTACTCC_HKVTWDSX3', '1275-GABA')]\n",
      "4321-GABA\n",
      "[('4321-GABA_TTGCCTAG-TAAGTGGT_HKVTWDSX3', '4321-GABA')]\n",
      "5606-GABA\n",
      "[('5606-GABA_GGCTTAAG-GGTCACGA_HKVTWDSX3', '5606-GABA')]\n",
      "1277-GLU\n",
      "[('1277-GLU_GGAGCGTC-GCACGGAC_HKVTWDSX3', '1277-GLU')]\n",
      "Hct15HBMA007-GABA\n",
      "[]\n",
      "5309-GABA\n",
      "[('5309-GABA_ATATGGAT-TAATACAG_HKVTWDSX3', '5309-GABA')]\n",
      "1134-GABA\n",
      "[('1134-GABA_GGTACCTT-GACGTCTT_HKVTWDSX3', '1134-GABA')]\n",
      "5451-GLU\n",
      "[('5451-GLU_GCAATGCA-AACGTTCC_HKVTWDSX3', '5451-GLU')]\n",
      "1105-GABA\n",
      "[('1105-GABA_AACTGTAG-TGCGGCGT_HKVTWDSX3', '1105-GABA')]\n",
      "4332-GLU\n",
      "[('4332-GLU_GTGAATAT-GAATGAGA_HKVTWDSX3', '4332-GLU')]\n",
      "198035-1765-GLU\n",
      "[('198035-1765-GLU_CTAGCGCT-GTCTACAC_HKVTWDSX3', '198035-1765-GLU')]\n",
      "5161-GLU\n",
      "[('5161-GLU_GAACCGCG-TAAGGTCA_HKVTWDSX3', '5161-GLU')]\n",
      "5617-GLU\n",
      "[('5617-GLU_TTACAGGA-GCTTGTCA_HKVTWDSX3', '5617-GLU')]\n",
      "HctZZNA014-GABA\n",
      "[]\n",
      "5918-GLU\n",
      "[('5918-GLU_GACGTCTT-GGTGAACC_HKVTWDSX3', '5918-GLU')]\n",
      "5606-GLU\n",
      "[('5606-GLU_AATCCGGA-AACTGTAG_HKVTWDSX3', '5606-GLU')]\n",
      "1848-GLU\n",
      "[('1848-GLU_GCGCTCTA-GTCGGAGC_HKVTWDSX3', '1848-GLU')]\n",
      "5787-GABA\n",
      "[('5787-GABA_TTCCTGTT-AAGATACT_HKVTWDSX3', '5787-GABA')]\n",
      "5451-GABA\n",
      "[('5451-GABA_ATGGCATG-GGTACCTT_HKVTWDSX3', '5451-GABA')]\n"
     ]
    }
   ],
   "source": [
    "prac = [i.split('_')[-1] for i in curr_dict['ID']]\n",
    "os.chdir('/datasets/Public_Datasets/Dracheva_PsychEncode_development/processed/rna_seq/')\n",
    "curr = glob.glob('*-*')\n",
    "for p in prac:\n",
    "    print(p)\n",
    "    print([(curr[i], p) for i in range(len(curr)) if p in curr[i]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'in_path': '/cndd3/dburrows/DATA/te/rna/PE.bam',\n",
       " 'out_path': '/cndd/dburrows/DATA/te/rna/PE.counts/RNA-mut',\n",
       " 'ID': ['Sample_5570-GABA',\n",
       "  'Sample_5086-GLU',\n",
       "  'Sample_1823-GABA',\n",
       "  'Sample_6285-GABA',\n",
       "  'Sample_1277-GABA',\n",
       "  'Sample_5976-GABA',\n",
       "  'Sample_5161-GABA',\n",
       "  'Sample_4411-GABA',\n",
       "  'Sample_4369-GLU',\n",
       "  'Sample_5326-GABA',\n",
       "  'Sample_HctYPA008-GABA',\n",
       "  'Sample_5309-GLU',\n",
       "  'Sample_4411-GLU',\n",
       "  'Sample_5871-GLU',\n",
       "  'Sample_5077-GABA',\n",
       "  'Sample_198034-1435-GABA',\n",
       "  'Sample_5086-GABA',\n",
       "  'Sample_Hct15HBNA032-GABA',\n",
       "  'Sample_5936-GLU',\n",
       "  'Sample_4337-GABA',\n",
       "  'Sample_1241-GLU',\n",
       "  'Sample_5184-GABA',\n",
       "  'Sample_5918-GABA',\n",
       "  'Sample_5643-GABA',\n",
       "  'Sample_5387-GLU',\n",
       "  'Sample_5976-GLU',\n",
       "  'Sample_4725-GLU',\n",
       "  'Sample_4428-GABA',\n",
       "  'Sample_1539-GLU',\n",
       "  'Sample_4332-GABA',\n",
       "  'Sample_4379-GABA',\n",
       "  'Sample_1848-GABA',\n",
       "  'Sample_179134-1401-GABA',\n",
       "  'Sample_Hct16HCQA020-GLU',\n",
       "  'Sample_5643-GLU',\n",
       "  'Sample_4321-GLU',\n",
       "  'Sample_5744-GABA',\n",
       "  'Sample_5401-GABA',\n",
       "  'Sample_5293-GABA',\n",
       "  'Sample_1133-GLU',\n",
       "  'Sample_1134-GLU',\n",
       "  'Sample_4425-GLU',\n",
       "  'Sample_3739-GLU',\n",
       "  'Sample_6285-GLU',\n",
       "  'Sample_6007-GABA',\n",
       "  'Sample_4425-GABA',\n",
       "  'Sample_1133-GABA',\n",
       "  'Sample_4414-GABA',\n",
       "  'Sample_5977-GLU',\n",
       "  'Sample_5977-GABA',\n",
       "  'Sample_Hct16HCQA020-GABA',\n",
       "  'Sample_4369-GABA',\n",
       "  'Sample_4337-GLU',\n",
       "  'Sample_HctZZNA014-GLU',\n",
       "  'Sample_5744-GLU',\n",
       "  'Sample_5446-GABA',\n",
       "  'Sample_5871-GABA',\n",
       "  'Sample_5570-GLU',\n",
       "  'Sample_6032-GABA',\n",
       "  'Sample_4379-GLU',\n",
       "  'Sample_Hct15HBMA007-GLU',\n",
       "  'Sample_1275-GLU',\n",
       "  'Sample_Hct17HEIA010-GLU',\n",
       "  'Sample_4545-GABA',\n",
       "  'Sample_198035-1765-GABA',\n",
       "  'Sample_HctYPA008-GLU',\n",
       "  'Sample_5617-GABA',\n",
       "  'Sample_6007-GLU',\n",
       "  'Sample_1535-GLU',\n",
       "  'Sample_Hct16HECA028-GABA',\n",
       "  'Sample_5936-GABA',\n",
       "  'Sample_5326-GLU',\n",
       "  'Sample_Hct17HEIA010-GABA',\n",
       "  'Sample_6032-GLU',\n",
       "  'Sample_198034-1435-GLU',\n",
       "  'Sample_5184-GLU',\n",
       "  'Sample_5293-GLU',\n",
       "  'Sample_1823-GLU',\n",
       "  'Sample_4725-GABA',\n",
       "  'Sample_Hct16HECA028-GLU',\n",
       "  'Sample_1105-GLU',\n",
       "  'Sample_4414-GLU',\n",
       "  'Sample_179134-1401-GLU',\n",
       "  'Sample_5401-GLU',\n",
       "  'Sample_4545-GLU',\n",
       "  'Sample_1648-GABA',\n",
       "  'Sample_5787-GLU',\n",
       "  'Sample_1648-GLU',\n",
       "  'Sample_5446-GLU',\n",
       "  'Sample_4428-GLU',\n",
       "  'Sample_1241-GABA',\n",
       "  'Sample_5387-GABA',\n",
       "  'Sample_Hct15HAWA031-GABA',\n",
       "  'Sample_Hct15HAWA031-GLU',\n",
       "  'Sample_Hct15HBNA032-GLU',\n",
       "  'Sample_5077-GLU',\n",
       "  'Sample_1539-GABA',\n",
       "  'Sample_1275-GABA',\n",
       "  'Sample_4321-GABA',\n",
       "  'Sample_5606-GABA',\n",
       "  'Sample_1277-GLU',\n",
       "  'Sample_Hct15HBMA007-GABA',\n",
       "  'Sample_5309-GABA',\n",
       "  'Sample_1134-GABA',\n",
       "  'Sample_5451-GLU',\n",
       "  'Sample_1105-GABA',\n",
       "  'Sample_4332-GLU',\n",
       "  'Sample_198035-1765-GLU',\n",
       "  'Sample_5161-GLU',\n",
       "  'Sample_5617-GLU',\n",
       "  'Sample_HctZZNA014-GABA',\n",
       "  'Sample_5918-GLU',\n",
       "  'Sample_5606-GLU',\n",
       "  'Sample_1848-GLU',\n",
       "  'Sample_5787-GABA',\n",
       "  'Sample_5451-GABA']}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19\n",
      "19\n",
      "19\n",
      "19\n",
      "18\n",
      "19\n",
      "19\n",
      "19\n",
      "19\n",
      "19\n",
      "19\n",
      "19\n",
      "19\n",
      "19\n",
      "18\n",
      "19\n",
      "18\n",
      "19\n",
      "19\n",
      "19\n",
      "19\n"
     ]
    }
   ],
   "source": [
    "os.chdir('/cndd3/mcuoco/projects/czi_diversity_rnate/results/merge_bam/')\n",
    "curr_l = glob.glob('**')\n",
    "bam_ll = []\n",
    "for curr in curr_l:\n",
    "    bam_l = glob.glob(curr+'/*.bam')\n",
    "    print(len(bam_l))\n",
    "    bam_ll = np.append(bam_ll, bam_l)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import glob\n",
    "os.chdir('/cndd3/dburrows/DATA/te/rna/PE.bam')\n",
    "l = glob.glob('*Samp**')\n",
    "for n in l:\n",
    "    os.mkdir('/cndd/dburrows/DATA/te/rna/PE.counts/ATEM/' + n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Sample_5570-GABA',\n",
       " 'Sample_5086-GLU',\n",
       " 'Sample_1823-GABA',\n",
       " 'Sample_6285-GABA',\n",
       " 'Sample_1277-GABA',\n",
       " 'Sample_5976-GABA',\n",
       " 'Sample_5161-GABA',\n",
       " 'Sample_4411-GABA',\n",
       " 'Sample_4369-GLU',\n",
       " 'Sample_5326-GABA',\n",
       " 'Sample_HctYPA008-GABA',\n",
       " 'Sample_5309-GLU',\n",
       " 'Sample_4411-GLU',\n",
       " 'Sample_5871-GLU',\n",
       " 'Sample_5077-GABA',\n",
       " 'Sample_198034-1435-GABA',\n",
       " 'Sample_5086-GABA',\n",
       " 'Sample_Hct15HBNA032-GABA',\n",
       " 'Sample_5936-GLU',\n",
       " 'Sample_4337-GABA',\n",
       " 'Sample_1241-GLU',\n",
       " 'Sample_5184-GABA',\n",
       " 'Sample_5918-GABA',\n",
       " 'Sample_5643-GABA',\n",
       " 'Sample_5387-GLU',\n",
       " 'Sample_5976-GLU',\n",
       " 'Sample_4725-GLU',\n",
       " 'Sample_4428-GABA',\n",
       " 'Sample_1539-GLU',\n",
       " 'Sample_4332-GABA',\n",
       " 'Sample_4379-GABA',\n",
       " 'Sample_1848-GABA',\n",
       " 'Sample_179134-1401-GABA',\n",
       " 'Sample_Hct16HCQA020-GLU',\n",
       " 'Sample_5643-GLU',\n",
       " 'Sample_4321-GLU',\n",
       " 'Sample_5744-GABA',\n",
       " 'Sample_5401-GABA',\n",
       " 'Sample_5293-GABA',\n",
       " 'Sample_1133-GLU',\n",
       " 'Sample_1134-GLU',\n",
       " 'Sample_4425-GLU',\n",
       " 'Sample_3739-GLU',\n",
       " 'Sample_6285-GLU',\n",
       " 'Sample_6007-GABA',\n",
       " 'Sample_4425-GABA',\n",
       " 'Sample_1133-GABA',\n",
       " 'Sample_4414-GABA',\n",
       " 'Sample_5977-GLU',\n",
       " 'Sample_5977-GABA',\n",
       " 'Sample_Hct16HCQA020-GABA',\n",
       " 'Sample_4369-GABA',\n",
       " 'Sample_4337-GLU',\n",
       " 'Sample_HctZZNA014-GLU',\n",
       " 'Sample_5744-GLU',\n",
       " 'Sample_5446-GABA',\n",
       " 'Sample_5871-GABA',\n",
       " 'Sample_5570-GLU',\n",
       " 'Sample_6032-GABA',\n",
       " 'Sample_4379-GLU',\n",
       " 'Sample_Hct15HBMA007-GLU',\n",
       " 'Sample_1275-GLU',\n",
       " 'Sample_Hct17HEIA010-GLU',\n",
       " 'Sample_4545-GABA',\n",
       " 'Sample_198035-1765-GABA',\n",
       " 'Sample_HctYPA008-GLU',\n",
       " 'Sample_5617-GABA',\n",
       " 'Sample_6007-GLU',\n",
       " 'Sample_1535-GLU',\n",
       " 'Sample_Hct16HECA028-GABA',\n",
       " 'Sample_5936-GABA',\n",
       " 'Sample_5326-GLU',\n",
       " 'Sample_Hct17HEIA010-GABA',\n",
       " 'Sample_6032-GLU',\n",
       " 'Sample_198034-1435-GLU',\n",
       " 'Sample_5184-GLU',\n",
       " 'Sample_5293-GLU',\n",
       " 'Sample_1823-GLU',\n",
       " 'Sample_4725-GABA',\n",
       " 'Sample_Hct16HECA028-GLU',\n",
       " 'Sample_1105-GLU',\n",
       " 'Sample_4414-GLU',\n",
       " 'Sample_179134-1401-GLU',\n",
       " 'Sample_5401-GLU',\n",
       " 'Sample_4545-GLU',\n",
       " 'Sample_1648-GABA',\n",
       " 'Sample_5787-GLU',\n",
       " 'Sample_1648-GLU',\n",
       " 'Sample_5446-GLU',\n",
       " 'Sample_4428-GLU',\n",
       " 'Sample_1241-GABA',\n",
       " 'Sample_5387-GABA',\n",
       " 'Sample_Hct15HAWA031-GABA',\n",
       " 'Sample_Hct15HAWA031-GLU',\n",
       " 'Sample_Hct15HBNA032-GLU',\n",
       " 'Sample_5077-GLU',\n",
       " 'Sample_1539-GABA',\n",
       " 'Sample_1275-GABA',\n",
       " 'Sample_4321-GABA',\n",
       " 'Sample_5606-GABA',\n",
       " 'Sample_1277-GLU',\n",
       " 'Sample_Hct15HBMA007-GABA',\n",
       " 'Sample_5309-GABA',\n",
       " 'Sample_1134-GABA',\n",
       " 'Sample_5451-GLU',\n",
       " 'Sample_1105-GABA',\n",
       " 'Sample_4332-GLU',\n",
       " 'Sample_198035-1765-GLU',\n",
       " 'Sample_5161-GLU',\n",
       " 'Sample_5617-GLU',\n",
       " 'Sample_HctZZNA014-GABA',\n",
       " 'Sample_5918-GLU',\n",
       " 'Sample_5606-GLU',\n",
       " 'Sample_1848-GLU',\n",
       " 'Sample_5787-GABA',\n",
       " 'Sample_5451-GABA']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "bed_pl = pd.read_csv(\"/cndd3/dburrows/DATA/te/gtf/bed/rmsk.hg38.filt-5ptrim.plus.sort.bed\", sep='\\t', header=None)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24038</td>\n",
       "      <td>24138</td>\n",
       "      <td>+</td>\n",
       "      <td>MIR</td>\n",
       "      <td>MIR</td>\n",
       "      <td>SINE</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26741</td>\n",
       "      <td>26841</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSp</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>31386</td>\n",
       "      <td>31486</td>\n",
       "      <td>+</td>\n",
       "      <td>AluJo</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32791</td>\n",
       "      <td>32891</td>\n",
       "      <td>+</td>\n",
       "      <td>MIR</td>\n",
       "      <td>MIR</td>\n",
       "      <td>SINE</td>\n",
       "      <td>214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39574</td>\n",
       "      <td>39674</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSx</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57200662</td>\n",
       "      <td>57200762</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSg7</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57201438</td>\n",
       "      <td>57201538</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSp</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57202215</td>\n",
       "      <td>57202315</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSz</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57202519</td>\n",
       "      <td>57202619</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSc</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57203511</td>\n",
       "      <td>57203611</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSz</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>277</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>548911 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           0         1         2  3       4    5     6    7\n",
       "0       chr1     24038     24138  +     MIR  MIR  SINE  211\n",
       "1       chr1     26741     26841  +   AluSp  Alu  SINE  263\n",
       "2       chr1     31386     31486  +   AluJo  Alu  SINE  297\n",
       "3       chr1     32791     32891  +     MIR  MIR  SINE  214\n",
       "4       chr1     39574     39674  +   AluSx  Alu  SINE  300\n",
       "...      ...       ...       ... ..     ...  ...   ...  ...\n",
       "548906  chrY  57200662  57200762  +  AluSg7  Alu  SINE  308\n",
       "548907  chrY  57201438  57201538  +   AluSp  Alu  SINE  305\n",
       "548908  chrY  57202215  57202315  +   AluSz  Alu  SINE  278\n",
       "548909  chrY  57202519  57202619  +   AluSc  Alu  SINE  308\n",
       "548910  chrY  57203511  57203611  +   AluSz  Alu  SINE  277\n",
       "\n",
       "[548911 rows x 8 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed_pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>family_id</th>\n",
       "      <th>class_id</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24038</td>\n",
       "      <td>24138</td>\n",
       "      <td>+</td>\n",
       "      <td>MIR</td>\n",
       "      <td>MIR</td>\n",
       "      <td>SINE</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26741</td>\n",
       "      <td>26841</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSp</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>31386</td>\n",
       "      <td>31486</td>\n",
       "      <td>+</td>\n",
       "      <td>AluJo</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32791</td>\n",
       "      <td>32891</td>\n",
       "      <td>+</td>\n",
       "      <td>MIR</td>\n",
       "      <td>MIR</td>\n",
       "      <td>SINE</td>\n",
       "      <td>214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39574</td>\n",
       "      <td>39674</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSx</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57200662</td>\n",
       "      <td>57200762</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSg7</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57201438</td>\n",
       "      <td>57201538</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSp</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57202215</td>\n",
       "      <td>57202315</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSz</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57202519</td>\n",
       "      <td>57202619</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSc</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57203511</td>\n",
       "      <td>57203611</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSz</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>277</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>548911 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chromosome     Start       End Strand gene_id family_id class_id  \\\n",
       "0            chr1     24038     24138      +     MIR       MIR     SINE   \n",
       "1            chr1     26741     26841      +   AluSp       Alu     SINE   \n",
       "2            chr1     31386     31486      +   AluJo       Alu     SINE   \n",
       "3            chr1     32791     32891      +     MIR       MIR     SINE   \n",
       "4            chr1     39574     39674      +   AluSx       Alu     SINE   \n",
       "...           ...       ...       ...    ...     ...       ...      ...   \n",
       "548906       chrY  57200662  57200762      +  AluSg7       Alu     SINE   \n",
       "548907       chrY  57201438  57201538      +   AluSp       Alu     SINE   \n",
       "548908       chrY  57202215  57202315      +   AluSz       Alu     SINE   \n",
       "548909       chrY  57202519  57202619      +   AluSc       Alu     SINE   \n",
       "548910       chrY  57203511  57203611      +   AluSz       Alu     SINE   \n",
       "\n",
       "        length  \n",
       "0          211  \n",
       "1          263  \n",
       "2          297  \n",
       "3          214  \n",
       "4          300  \n",
       "...        ...  \n",
       "548906     308  \n",
       "548907     305  \n",
       "548908     278  \n",
       "548909     308  \n",
       "548910     277  \n",
       "\n",
       "[548911 rows x 8 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed_pl.columns =['Chromosome', 'Start', 'End', 'Strand', 'gene_id', 'family_id', 'class_id', 'length']\n",
    "bed_pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>family_id</th>\n",
       "      <th>class_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24038</td>\n",
       "      <td>24138</td>\n",
       "      <td>+</td>\n",
       "      <td>MIR</td>\n",
       "      <td>MIR</td>\n",
       "      <td>SINE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26741</td>\n",
       "      <td>26841</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSp</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>31386</td>\n",
       "      <td>31486</td>\n",
       "      <td>+</td>\n",
       "      <td>AluJo</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32791</td>\n",
       "      <td>32891</td>\n",
       "      <td>+</td>\n",
       "      <td>MIR</td>\n",
       "      <td>MIR</td>\n",
       "      <td>SINE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39574</td>\n",
       "      <td>39674</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSx</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548906</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57200662</td>\n",
       "      <td>57200762</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSg7</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548907</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57201438</td>\n",
       "      <td>57201538</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSp</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548908</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57202215</td>\n",
       "      <td>57202315</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSz</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548909</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57202519</td>\n",
       "      <td>57202619</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSc</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548910</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57203511</td>\n",
       "      <td>57203611</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSz</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>548911 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chromosome     Start       End Strand gene_id family_id class_id\n",
       "0            chr1     24038     24138      +     MIR       MIR     SINE\n",
       "1            chr1     26741     26841      +   AluSp       Alu     SINE\n",
       "2            chr1     31386     31486      +   AluJo       Alu     SINE\n",
       "3            chr1     32791     32891      +     MIR       MIR     SINE\n",
       "4            chr1     39574     39674      +   AluSx       Alu     SINE\n",
       "...           ...       ...       ...    ...     ...       ...      ...\n",
       "548906       chrY  57200662  57200762      +  AluSg7       Alu     SINE\n",
       "548907       chrY  57201438  57201538      +   AluSp       Alu     SINE\n",
       "548908       chrY  57202215  57202315      +   AluSz       Alu     SINE\n",
       "548909       chrY  57202519  57202619      +   AluSc       Alu     SINE\n",
       "548910       chrY  57203511  57203611      +   AluSz       Alu     SINE\n",
       "\n",
       "[548911 rows x 7 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#drop last columns\n",
    "bed_pl = bed_pl.drop(columns=['length'])\n",
    "bed_pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Check that all files are present\n",
    "import glob\n",
    "import os\n",
    "\n",
    "full_l = os.listdir('/cndd2/Public_Datasets/Dracheva_PsychEncode_development/raw_May2022/') \n",
    "os.chdir('/cndd3/dburrows/DATA/te/rna/PE.bam/')\n",
    "check_l = glob.glob('*Samp*')\n",
    "\n",
    "import numpy as np\n",
    "not_int = np.setxor1d(check_l, full_l)\n",
    "assert len(full_l) == len(check_l)\n",
    "assert len(not_int) == 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "STAR --genomeDir /cndd2/jchien/iGenome/STAR_gencode_v37 --readFilesIn  R1-merge.fastq.gz R1-merge.fastq.gz --outSAMunmapped None --outFilterType BySJout --outSAMattributes All --outFilterMultimapNmax 100 --outFilterMismatchNmax 999 --outFilterMismatchNoverReadLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesCommand zcat --runThreadN 16--genomeLoad LoadAndKeep --limitBAMsortRAM 10000000000 --outSAMtype BAM SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --winAnchorMultimapNmax 200 --outMultimapperOrder Random --outSAMmultNmax -1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check which files went to completion\n",
    "import glob\n",
    "os.chdir('/cndd/dburrows/DATA/te/rna/PE.counts/ATEM/')\n",
    "curr_ = glob.glob('*Sa*')\n",
    "for c in curr_:\n",
    "    os.chdir(c)\n",
    "    atem = glob.glob('/cndd/dburrows/DATA/te/rna/PE.counts/ATEM/' + c + '/*ATEM*')\n",
    "    log = glob.glob('/cndd/dburrows/DATA/te/rna/PE.counts/ATEM/' + c + '/*log*')\n",
    "    if not atem or not log:\n",
    "        print(c)\n",
    "    if log:\n",
    "        if 'All reads accounted for' not in np.load(log[0]):\n",
    "            print(c)\n",
    "\n",
    "    os.chdir('..')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
