{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3.9.12 (main, Apr  5 2022, 06:56:58) \\n[GCC 7.5.0]'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#CONDA ENV base (python 3.9.12)\n",
    "#Import packages\n",
    "#---------------------------------------\n",
    "import json\n",
    "import pandas as pd\n",
    "import pyranges as pr\n",
    "import sys\n",
    "\n",
    "#Import your modules\n",
    "#---------------------------------------\n",
    "sys.path.insert(1, '/cndd3/dburrows/CODE/te_ageing/')\n",
    "import te_rna_f as te\n",
    "sys.path.insert(1, '/cndd3/dburrows/CODE/admin_tools/')\n",
    "from admin_tools import admin_functions as adm\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Read in required files for filtering\n",
    "js = json.load(open('config.json'))\n",
    "bed_pl = pd.read_csv(js['bed_plus_path'], sep='\\t', header=None)\n",
    "bed_pl.columns =['Chromosome', 'Start', 'End', 'Strand', 'transcript_id', 'gene_id', 'family_id', 'class_id']\n",
    "bed_mi = pd.read_csv(js['bed_minus_path'],sep='\\t', header=None)\n",
    "bed_mi.columns =['Chromosome', 'Start', 'End', 'Strand', 'transcript_id', 'gene_id', 'family_id', 'class_id']\n",
    "\n",
    "bam_pl = pr.read_bam(snakemake.input.bam_pl, as_df=True) \n",
    "bam_mi = pr.read_bam(snakemake.input.bam_mi, as_df=True) \n",
    "#Swap Start + End for minus strand\n",
    "bam_mi['Start'], bam_mi['End'] = bam_mi['End'], bam_mi['Start']\n",
    "\n",
    "#File checks\n",
    "assert sum(bam_pl['Strand'] == '+') == len(bam_pl), 'Some non plus strands assigned to plus bam'\n",
    "assert sum(bam_mi['Strand'] == '-') == len(bam_mi), 'Some non minus strands assigned to minus bam'\n",
    "assert sum(bed_pl['Strand'] == '+') == len(bed_pl), 'Some non plus strands assigned to plus bed'\n",
    "assert sum(bed_mi['Strand'] == '-') == len(bed_mi), 'Some non minus strands assigned to minus bed'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Define + and - strand files\n",
    "pl_pars = [bed_pl, bam_pl, snakemake.input.meta_pl, 'plus'] \n",
    "mi_pars = [bed_mi, bam_mi, snakemake.input.meta_mi, 'minus'] \n",
    "par_list = [pl_pars, mi_pars]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligning to chromosome chr1 for plus strand\n",
      "Aligning to chromosome chr10 for plus strand\n",
      "Aligning to chromosome chr11 for plus strand\n",
      "Aligning to chromosome chr12 for plus strand\n",
      "Aligning to chromosome chr13 for plus strand\n",
      "Aligning to chromosome chr14 for plus strand\n",
      "Aligning to chromosome chr15 for plus strand\n",
      "Aligning to chromosome chr16 for plus strand\n",
      "Aligning to chromosome chr17 for plus strand\n",
      "Aligning to chromosome chr18 for plus strand\n",
      "Aligning to chromosome chr19 for plus strand\n",
      "Aligning to chromosome chr2 for plus strand\n",
      "Aligning to chromosome chr20 for plus strand\n",
      "Aligning to chromosome chr21 for plus strand\n",
      "Aligning to chromosome chr22 for plus strand\n",
      "Aligning to chromosome chr3 for plus strand\n",
      "Aligning to chromosome chr4 for plus strand\n",
      "Aligning to chromosome chr5 for plus strand\n",
      "Aligning to chromosome chr6 for plus strand\n",
      "Aligning to chromosome chr7 for plus strand\n",
      "Aligning to chromosome chr8 for plus strand\n",
      "Aligning to chromosome chr9 for plus strand\n",
      "Aligning to chromosome chrX for plus strand\n",
      "Aligning to chromosome chrY for plus strand\n",
      "Aligning to chromosome chr1 for minus strand\n",
      "Aligning to chromosome chr10 for minus strand\n",
      "Aligning to chromosome chr11 for minus strand\n",
      "Aligning to chromosome chr12 for minus strand\n",
      "Aligning to chromosome chr13 for minus strand\n",
      "Aligning to chromosome chr14 for minus strand\n",
      "Aligning to chromosome chr15 for minus strand\n",
      "Aligning to chromosome chr16 for minus strand\n",
      "Aligning to chromosome chr17 for minus strand\n",
      "Aligning to chromosome chr18 for minus strand\n",
      "Aligning to chromosome chr19 for minus strand\n",
      "Aligning to chromosome chr2 for minus strand\n",
      "Aligning to chromosome chr20 for minus strand\n",
      "Aligning to chromosome chr21 for minus strand\n",
      "Aligning to chromosome chr22 for minus strand\n",
      "Aligning to chromosome chr3 for minus strand\n",
      "Aligning to chromosome chr4 for minus strand\n",
      "Aligning to chromosome chr5 for minus strand\n",
      "Aligning to chromosome chr6 for minus strand\n",
      "Aligning to chromosome chr7 for minus strand\n",
      "Aligning to chromosome chr8 for minus strand\n",
      "Aligning to chromosome chr9 for minus strand\n",
      "Aligning to chromosome chrX for minus strand\n",
      "Aligning to chromosome chrY for minus strand\n"
     ]
    }
   ],
   "source": [
    "pd.options.mode.chained_assignment = None  # default='warn'\n",
    "count_df = pd.DataFrame() #empty count matrix\n",
    "bam_ll = [[],[]] #empty list of lists to store curr_bam indices\n",
    "\n",
    "# Filter out reads that do not overlap with 5' portion of insertion\n",
    "for x,par in enumerate(par_list):\n",
    "    curr_bed = par[0]\n",
    "    curr_bam = par[1]\n",
    "    curr_name = pd.read_csv(par[2], sep='\\t', header=None)\n",
    "    assert len(curr_bam) == len(curr_name), 'Bam and metadata files not the same length'\n",
    "    curr_bam['UMI']=curr_name[0].values #Add UMI column to bam file\n",
    "\n",
    "    #Loop through each chromosome\n",
    "    chr_unq = np.unique(curr_bam['Chromosome'].values)\n",
    "    for i,chr in enumerate(chr_unq):\n",
    "        print('Aligning to chromosome ' + chr + ' for ' + par[3] + ' strand')\n",
    "        \n",
    "        #Slice bed/bam files by chromosome\n",
    "        chr_bam = curr_bam[curr_bam['Chromosome'] == chr]\n",
    "        chr_bed = curr_bed[curr_bed['Chromosome'] == chr]\n",
    "        count_df, bam_ll[x] = te.five_prime_align(chr_bam, chr_bed, count_df, bam_ll[x])\n",
    "\n",
    "#Add in CPMs as a column\n",
    "total_reads = pd.read_csv(snakemake.inputs.n_reads, sep=\" \", header=None)[0].values[0] \n",
    "count_df['CPM'] = count_df['Count'].values / total_reads * 1000000 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>family_id</th>\n",
       "      <th>class_id</th>\n",
       "      <th>Count</th>\n",
       "      <th>CPM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>59460</th>\n",
       "      <td>chr1</td>\n",
       "      <td>240195735</td>\n",
       "      <td>240195755</td>\n",
       "      <td>+</td>\n",
       "      <td>AluJo_dup6673</td>\n",
       "      <td>AluJo</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.003852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21778</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1357526</td>\n",
       "      <td>1357546</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSx1_dup79</td>\n",
       "      <td>AluSx1</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.003852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49347</th>\n",
       "      <td>chr1</td>\n",
       "      <td>172601207</td>\n",
       "      <td>172601227</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSz_dup6301</td>\n",
       "      <td>AluSz</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.003852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34049</th>\n",
       "      <td>chr1</td>\n",
       "      <td>46158806</td>\n",
       "      <td>46158826</td>\n",
       "      <td>+</td>\n",
       "      <td>AluSx3_dup949</td>\n",
       "      <td>AluSx3</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.003852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55356</th>\n",
       "      <td>chr1</td>\n",
       "      <td>216510603</td>\n",
       "      <td>216510623</td>\n",
       "      <td>+</td>\n",
       "      <td>AluY_dup7600</td>\n",
       "      <td>AluY</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.003852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473243</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14747792</td>\n",
       "      <td>14747812</td>\n",
       "      <td>-</td>\n",
       "      <td>AluJr_dup46073</td>\n",
       "      <td>AluJr</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.003852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472889</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12405168</td>\n",
       "      <td>12405188</td>\n",
       "      <td>-</td>\n",
       "      <td>AluJr_dup45997</td>\n",
       "      <td>AluJr</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.003852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473140</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13759996</td>\n",
       "      <td>13760016</td>\n",
       "      <td>-</td>\n",
       "      <td>AluSx_dup63231</td>\n",
       "      <td>AluSx</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.003852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472578</th>\n",
       "      <td>chrY</td>\n",
       "      <td>8521782</td>\n",
       "      <td>8521802</td>\n",
       "      <td>-</td>\n",
       "      <td>AluYe5_dup763</td>\n",
       "      <td>AluYe5</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.003852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473772</th>\n",
       "      <td>chrY</td>\n",
       "      <td>18976012</td>\n",
       "      <td>18976032</td>\n",
       "      <td>-</td>\n",
       "      <td>AluSx1_dup62835</td>\n",
       "      <td>AluSx1</td>\n",
       "      <td>Alu</td>\n",
       "      <td>SINE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.003852</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>273309 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chromosome      Start        End Strand    transcript_id gene_id  \\\n",
       "59460        chr1  240195735  240195755      +    AluJo_dup6673   AluJo   \n",
       "21778        chr1    1357526    1357546      +     AluSx1_dup79  AluSx1   \n",
       "49347        chr1  172601207  172601227      +    AluSz_dup6301   AluSz   \n",
       "34049        chr1   46158806   46158826      +    AluSx3_dup949  AluSx3   \n",
       "55356        chr1  216510603  216510623      +     AluY_dup7600    AluY   \n",
       "...           ...        ...        ...    ...              ...     ...   \n",
       "473243       chrY   14747792   14747812      -   AluJr_dup46073   AluJr   \n",
       "472889       chrY   12405168   12405188      -   AluJr_dup45997   AluJr   \n",
       "473140       chrY   13759996   13760016      -   AluSx_dup63231   AluSx   \n",
       "472578       chrY    8521782    8521802      -    AluYe5_dup763  AluYe5   \n",
       "473772       chrY   18976012   18976032      -  AluSx1_dup62835  AluSx1   \n",
       "\n",
       "       family_id class_id  Count       CPM  \n",
       "59460        Alu     SINE    1.0  0.003852  \n",
       "21778        Alu     SINE    1.0  0.003852  \n",
       "49347        Alu     SINE    1.0  0.003852  \n",
       "34049        Alu     SINE    1.0  0.003852  \n",
       "55356        Alu     SINE    1.0  0.003852  \n",
       "...          ...      ...    ...       ...  \n",
       "473243       Alu     SINE    1.0  0.003852  \n",
       "472889       Alu     SINE    1.0  0.003852  \n",
       "473140       Alu     SINE    1.0  0.003852  \n",
       "472578       Alu     SINE    1.0  0.003852  \n",
       "473772       Alu     SINE    1.0  0.003852  \n",
       "\n",
       "[273309 rows x 10 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Save counts matrix\n",
    "count_df.to_csv(snakemake.output.count_mat, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Obtain start positions\n",
    "#Make txt file of start sites to remove\n",
    "pl_umi=pd.read_csv(snakemake.input.meta_pl, sep='\\t', header=None).iloc[np.setxor1d(np.arange(0,len(bam_pl)) , bam_ll[0].astype(int))]\n",
    "mi_umi=pd.read_csv(snakemake.input.meta_mi, sep='\\t', header=None).iloc[np.setxor1d(np.arange(0,len(bam_mi)) , bam_ll[1].astype(int))]\n",
    "np.savetxt(snakemake.output.meta_pl,  pl_umi, fmt='%s')\n",
    "np.savetxt(snakemake.output.meta_mi,  mi_umi, fmt='%s')\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
